src/hotspot/share/compiler/compilerDefinitions.cpp

```diff
@@ -127,7 +127,14 @@ intx CompilerConfig::scaled_compile_threshold(intx threshold, double scale) {
   if (scale == 1.0 || scale < 0.0) {
     return threshold;
   } else {
-    return (intx)(threshold * scale);
+    double v = threshold * scale;
+    if (v > max_intx) {
+      return max_intx;
+    } else if (v < min_intx) {
```

Comment:
This check is meaningless since `v` can't be negative. If you want you can add assert.

Reply:
I removed the `min_intx` check and added the asserts. I ran with the following tests and it didn't trigger the asserts. I'll rerun tiers 1-3 again.

```
java -XX:CompileThreshold=-1         -XX:CompileThresholdScaling=1 -version
java -XX:Tier0InvokeNotifyFreqLog=-2 -XX:CompileThresholdScaling=1 -version
java -XX:Tier4BackEdgeThreshold=-2   -XX:CompileThresholdScaling=1 -version
```

Reply:
Yes, there is `range()` checks for these flags declarations:
```
java -XX:CompileThreshold=-1         -XX:CompileThresholdScaling=1 -version
OnStackReplacePercentage cannot be validated because CompileThreshold value is invalid
CompileThreshold (-1) must be between 0 and 1073741823
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```

You can't specify negative Threshold values on command line. The assert is for some future case if we pass incorrect value.